tests/unit/iotics/lib/identity/register/test_document_builder.py

```diff
@@ -134,6 +134,23 @@ def test_can_build_a_register_doc_from_dict(minimal_doc, full_doc, is_minimal):
     assert new_doc.auth_delegation_proof == doc.auth_delegation_proof
 
 
+@pytest.mark.parametrize('is_minimal', (True, False))
+def test_can_build_a_register_doc_from_dict_absent_auth(minimal_doc, full_doc, is_minimal):
```

Comment:
this one is not testing the "absence of auth_keys or public keys or delegation" fields.
Because to_dict() will set those fields with an explicit empty list

This test could be:
```
def test_can_build_a_register_doc_from_minimal_dict(minimal_doc):
    doc_as_dict = minimal_doc.to_dict()
    doc_as_dict.pop('authentication')
    doc_as_dict.pop('delegateControl')
    doc_as_dict.pop('delegateAuthentication')

    new_doc = RegisterDocumentBuilder().build_from_dict(doc_as_dict)
    assert new_doc.did == doc.did
    assert new_doc.purpose == doc.purpose
    assert new_doc.proof == doc.proof
    assert new_doc.revoked == doc.revoked
    assert new_doc.spec_version == doc.spec_version
    assert new_doc.metadata == doc.metadata
    assert new_doc.public_keys == doc.public_keys
    assert not new_doc.auth_keys
    assert not new_doc.control_delegation_proof 
    assert not new_doc.auth_delegation_proof 
```